src/elements/Button/Button.js

```diff
@@ -1,33 +1,136 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+
 import META from '../../utils/Meta'
+import * as sui from '../../utils/semanticUtils'
+import {
+  iconPropRenderer,
+  useKeyOrValueAndKey,
+  getUnhandledProps,
+  useValueAndKey,
+  useKeyOnly,
+} from '../../utils/propUtils'
+import ButtonGroup from './ButtonGroup'
+
+/**
+ * Button
+ */
+function Button(props) {
+  const {
+    children, color, size, className, animated, floated, labeled, icon, inverted, fluid, basic, active, disabled,
+    loading, circular, compact, attached, toggle, type,
+  } = props
+
+  const classes = cx('ui',
+    icon && 'icon',
+    color,
+    size,
+    useKeyOnly(active, 'active'),
+    useKeyOnly(animated, 'animated'),
+    useKeyOrValueAndKey(attached, 'attached'),
+    useKeyOnly(basic, 'basic'),
+    useKeyOnly(circular, 'circular'),
+    useKeyOnly(compact, 'compact'),
+    useKeyOnly(disabled, 'disabled'),
+    useValueAndKey(floated, 'floated'),
+    useKeyOnly(fluid, 'fluid'),
+    useKeyOnly(inverted, 'inverted'),
+    useKeyOrValueAndKey(labeled, 'labeled'),
+    useKeyOnly(loading, 'loading'),
+    useKeyOnly(toggle, 'toggle'),
+    'button',
+    className
+  )
+
+  const rest = getUnhandledProps(Button, props)
+
+  return (
+    <button type={type} className={classes} {...rest}>
```

Comment:
When the `attached` prop is used, the button component needs to be a `div`.  This fixes #46.  The CSS does not work on a `button` element for attached buttons.

```html
<div class="ui top attached button" tabindex="0">Top</div>
<div class="ui attached segment">
  <p></p>
</div>
<div class="ui bottom attached button" tabindex="0">Bottom</div>
```

Also, the `tabIndex={0}` prop will need to be added so users can still tab to the button `div`.